internal/metamorph/processor.go

```diff
@@ -857,3 +878,25 @@ func callbackExists(callback store.Callback, data *store.Data) bool {
 	}
 	return false
 }
+
+func (p *Processor) saveTxToCache(hash *chainhash.Hash) {
+	if p.cacheStore.IsShared() {
+		_ = p.cacheStore.MapSet(CacheRegisteredTxsHash, hash.String(), []byte("1"))
+	}
+}
+
+func (p *Processor) txFoundInCache(hash *chainhash.Hash) (found bool) {
+	if p.cacheStore.IsShared() {
+		value, _ := p.cacheStore.MapGet(CacheRegisteredTxsHash, hash.String())
+		return value != nil
+	}
+
+	// if we don't have a shared cache running, treat all transactions as found
```

Comment:
I think if the cache is not shared (in-memory) then we could still use this optimisation.
The only difference will be that if another metamorph sees a transaction it will not be updated to seen on network.

Reply:
You're right. In case the metamorph instance goes down/unresponsive/overloaded, we will lose statuses up until `SEEN_ON_NETWORK`, but then it should still get updated to `MINED`. Worst case scenario, the broken metamorph instance will have its transactions unlocked by k8s-watcher and they can be picked up by a different instance.

Okay, removed the condition.